test/distributed/org/apache/cassandra/fuzz/sai/SingleNodeSAITestBase.java

```diff
@@ -193,33 +232,50 @@ private void basicSaiTest(EntropySource rng, SchemaSpec schema)
             if (rng.nextFloat() > 0.9995f)
                 history.deletePartition(partitionIndex);
 
-            if (i % REPAIR_SKIP == 0)
-                history.custom(() -> repair(schema), "Repair");
-            else if (i % FLUSH_SKIP == 0)
+            if (i % FLUSH_SKIP == 0)
                 history.custom(() -> flush(schema), "Flush");
             else if (i % COMPACTION_SKIP == 0)
                 history.custom(() -> compact(schema), "Compact");
+            else if (i % repairSkip == 0)
+                history.custom(() -> repair(schema), "Repair");
 
-            if (i > 0 && i % 1000 == 0)
+            if (i > 0 && i % VALIDATION_SKIP == 0)
             {
-                for (int j = 0; j < 5; j++)
+                for (int j = 0; j < QUERIES_PER_VALIDATION; j++)
                 {
-                    List<IdxRelation> regularRelations = HistoryBuilderHelper.generateValueRelations(rng, schema.regularColumns.size(),
-                                                                                                     column -> Math.min(schema.valueGenerators.regularPopulation(column), UNIQUE_CELL_VALUES));
-                    List<IdxRelation> staticRelations = HistoryBuilderHelper.generateValueRelations(rng, schema.staticColumns.size(),
-                                                                                                    column -> Math.min(schema.valueGenerators.staticPopulation(column), UNIQUE_CELL_VALUES));
-                    history.select(pkGen.generate(rng),
-                                   HistoryBuilderHelper.generateClusteringRelations(rng, schema.clusteringKeys.size(), ckGen).toArray(new IdxRelation[0]),
-                                   regularRelations.toArray(new IdxRelation[regularRelations.size()]),
-                                   staticRelations.toArray(new IdxRelation[staticRelations.size()]));
+                    List<IdxRelation> regularRelations = 
+                            HistoryBuilderHelper.generateValueRelations(rng,
+                                                                        schema.regularColumns.size(),
+                                                                        column -> Math.min(schema.valueGenerators.regularPopulation(column), UNIQUE_CELL_VALUES),
+                                                                        eqOnlyRegularColumns);
```

Comment:
@ifesdjeen given the current pattern, wouldn't this be better as a `Predicate` rather than a set?

We have `Function<Integer, Integer> population`, so maybe adding

```
IntPredicate eqOnly
```